src/View/Helper/IdentityHelper.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         1.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authentication\View\Helper;
+
+use Authentication\IdentityInterface;
+use Cake\ORM\Entity;
+use Cake\Utility\Hash;
+use Cake\View\Helper;
+use RuntimeException;
+
+/**
+ * Identity Helper
+ *
+ * A convenience helper to access the identity data
+ */
+class IdentityHelper extends Helper
+{
+    /**
+     * User data
+     *
+     * @array
+     */
+    protected $_userData = [];
+
+    /**
+     * Identity Object
+     *
+     * @var null|\Authentication\IdentityInterface
+     */
+    protected $_identity;
+
+    /**
+     * Constructor hook method.
+     *
+     * Implement this method to avoid having to overwrite the constructor and call parent.
+     *
+     * @param array $config The configuration settings provided to this helper.
+     * @return void
+     */
+    public function initialize(array $config)
+    {
+        $this->_identity = $this->getView()->request->getAttribute('identity');
+
+        if (empty($this->_identity)) {
+            return;
+        }
+
+        if (!$this->_identity instanceof IdentityInterface) {
+            throw new RuntimeException(sprintf('Identity found in request does not implement %s', IdentityInterface::class));
+        }
+
+        $this->_userData = $this->_identity->getOriginalData();
+
+        if ($this->_userData instanceof Entity) {
+            $this->_userData = $this->_userData->toArray();
+        }
+    }
+
+    /**
+     * Gets the id of the current logged in identity
+     *
+     * @return int|null|string
+     */
+    public function getId()
+    {
+        if (empty($this->_identity)) {
+            return null;
+        }
+
+        return $this->_identity->getIdentifier();
+    }
+
+    /**
+     * Checks if a user is logged in
+     *
+     * @return bool
+     */
+    public function isLoggedIn()
+    {
+        return !empty($this->_identity);
+    }
+
+    /**
+     * This check can be used to tell if a record that belongs to some user is the
+     * current logged in user
+     *
+     * @param int|string $id Identity id to check against
+     * @param string $field Name of the field in the identity data to check against, id by default
+     * @return bool
+     */
+    public function is($id, $field = 'id')
```

Comment:
This method doesn't look like it should belong here. Checking if one user is another user is authorization. We didn't want to mix those concepts.

Reply:
So you want a full powered authorization plugin instead of this convenient check? This is pretty much all you need to conditionally show if the user is logged in or not.

Reply:
Is this method for checking whether a user is logged in? Docs say otherwise.

Reply:
My bad wording: You have pretty often the case when you want to check if an user can see something or not. This might depend on another field as well. The method provides an easy way to do that, that is intuitive accessible and doesn't require a second plugin nor knowledge about it. We could drop the whole helper and just add getIdentity() to the view class directly?

Reply:
Should we be guiding people to use Authorization policies in those scenarios? This method covers a very simple use case, but I feel like we should be recommending people with more complex requirements use the authorization plugin.

Reply:
Lets stick a comment in the doc block and recommend it there?

Reply:
That works for me.

Reply:
sounds good.

Reply:
@markstory can we merge it now? I've added the doc block.